### PR TITLE
Sessions: initial setup

### PR DESCRIPTION
**Pull request by jreij**

### CODE REVIEW

File: sessions-core/proguard-rules.pro [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Add project specific ProGuard rules here.
```
Comment (OscarSpruit) [inline sessions-core/proguard-rules.pro:1]: Noticed that most of these files are empty. We might as well remove them, to decrease some clutter

File: example-app/src/main/java/com/adyen/checkout/example/data/api/CheckoutApiService.kt [hunk 2]
```diff
@@ -30,6 +32,10 @@ interface CheckoutApiService {
         }
     }
 
+    @Headers(BuildConfig.API_KEY_HEADER_NAME + ":" + BuildConfig.CHECKOUT_API_KEY)
+    @POST("sessions")
+    fun sessionsAsync(@Body sessionRequest: SessionRequest): Deferred<Response<Session>>
```
[the remark this answers is not in the document]
Comment (jreij) [reply example-app/src/main/java/com/adyen/checkout/example/data/api/CheckoutApiService.kt:37]: Good catch 👍 
We have a bunch of other API calls like that though so maybe let's do it as a separate refactoring in another PR?

File: sessions-core/proguard-rules.pro (hunk 1 above)
Comment (jreij) [reply sessions-core/proguard-rules.pro:1]: @caiofaustino are these `proguard-rules.pro` files needed for anything? Or just the `consumer-rules.pro`?

File: example-app/src/main/java/com/adyen/checkout/example/data/api/CheckoutApiService.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (OscarSpruit) [reply example-app/src/main/java/com/adyen/checkout/example/data/api/CheckoutApiService.kt:37]: Sure, then it's at least consistent for now 😄

File: sessions-core/proguard-rules.pro (hunk 1 above)
Comment (caiofaustino) [reply sessions-core/proguard-rules.pro:1]: Yeah those will only be used if we decide to ship the SDK already obfuscated, but I don't think that will ever be the case. So feel free to remove them of you want.

**Pull request merged**